src/generators/non_gradient_based/feature_tweak/feature_tweak.jl

```diff
@@ -39,6 +39,21 @@ function feature_tweaking(
     x::AbstractArray,
     target::RawTargetType,
 )
+    if !(M isa Models.TreeModel)
+        throw(
+            ArgumentError(
+                "The feature tweak generator is only available for tree-based models."
+            ),
+        )
+    end
```

Comment:
this condition is never entered as the function parameters already require `M::Models.TreeModel` (line 38)

Reply:
Thanks, my bad. Will remove this condition.